cpp/open3d/t/geometry/PointCloud.cpp

```diff
@@ -259,8 +259,13 @@ open3d::geometry::PointCloud PointCloud::ToLegacyPointCloud() const {
                 core::eigen_converter::TensorToEigenVector3dVector(GetPoints());
     }
     if (HasPointColors()) {
-        pcd_legacy.colors_ = core::eigen_converter::TensorToEigenVector3dVector(
-                GetPointColors());
+        auto colors = GetPointColors();
+        if (colors.GetDtype() == core::Dtype::UInt8) {
+            colors = colors.To(core::Dtype::Float64) / 255.0;
+        }
+
```

Comment:
Do we need to check and convert for `UInt16` also ?

Reply:
We probably also do not want to hard-code the constant `255.0`. So something like `static_cast<double>(std::numeric_limits<uint8_t>::max())` would be much better to understand. This would also be more generic with respect to other types like `UInt16`, etc.